src/main/java/net/dv8tion/jda/internal/entities/EntitlementImpl.java

```diff
@@ -0,0 +1,107 @@
+package net.dv8tion.jda.internal.entities;
+
+import net.dv8tion.jda.api.entities.Entitlement;
+import net.dv8tion.jda.api.utils.data.DataObject;
+
+import javax.annotation.Nullable;
+import java.time.OffsetDateTime;
+
+public class EntitlementImpl implements Entitlement
+{
+    private String id;
+    private String skuId;
+    private String applicationId;
+    @Nullable
+    private String userId;
+    @Nullable
+    private String guildId;
+    private Integer type;
+    private Boolean deleted;
+    @Nullable
+    private OffsetDateTime startsAt;
+    @Nullable
+    private OffsetDateTime endsAt;
+
+    public EntitlementImpl(String id, String skuId, String applicationId, @Nullable String userId, @Nullable String guildId, Integer type, Boolean deleted, @Nullable OffsetDateTime startsAt, @Nullable OffsetDateTime endsAt)
+    {
+        this.id = id;
+        this.skuId = skuId;
+        this.applicationId = applicationId;
+        this.userId = userId;
+        this.guildId = guildId;
+        this.type = type;
+        this.deleted = deleted;
+        this.startsAt = startsAt;
+        this.endsAt = endsAt;
+    }
+
+    public EntitlementImpl(DataObject data) {
```

Comment:
```suggestion
    public EntitlementImpl(DataObject data)
    {
```

Reply:
We generally don't do parsing in constructors. We almost always have a method in EntityBuilder.